### ISSUE TITLE
brace-style needs an option for arrow function shorthand

### ISSUE THREAD

**Issue by yaycmyk** [eslint/eslint#2744]
In my project I'd like to force linebreaks for braced-statements in all cases except arrow function shorthand.

For example:

```js
// bad
function() { blah(); }

let foo = baz((prop) => {
    prop + 1
});

// good
function() {
    blah();
}

let foo = bar((baz) => { baz + 1 });
```

Is there a way to achieve this with the current ruleset?

**Comment by nzakas** [comment]
Is much easier for us to discuss this of you include how you configured the rule, what ESLint outputs with that configuration, and what you'd like it to output.

**Comment by gyandeeps** [comment]
@yaycmyk are you still interested in this and if yes then please provide more info.

**Comment by yaycmyk** [comment]
@gyandeeps the config I'm going for is 
```js
{
    brace-style: [2, "1tbs", {allowSingleLine: false, allowSingleLineArrowFunctions: true}]
}
```

So basically, the only allowed single-line setup is a one-liner arrow function such as my original example.

**Comment by nzakas** [comment]
In that case, why not omit the braces?

**Comment by gyandeeps** [comment]
@yaycmyk waiting for feedback/answer

**Comment by yaycmyk** [comment]
@nzakas's suggestion is sufficient for the use case - thanks

**Comment by tomasz-nowakowski-stx** [comment]
@nzakas if you omit the braces you will return value of `baz + 1`

It would be nice to have `allowSingleLineArrowFunctions` rule for simple setters:
`const setFoo = foo => { this.foo }`
